include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BATTERY
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <device.h>
+
+/*
+ * Sometimes we have hardware to detect battery present, sometimes we have to
+ * wait until we've been able to talk to the battery.
+ */
+enum battery_present {
+	BATTERY_PRESENCE_UNINITIALIZED = -1,
+	BATTERY_PRESENCE_NO = 0,
+	BATTERY_PRESENCE_YES = 1,
+	BATTERY_PRESENCE_UNKNOWN,
```

Comment:
Would there be benefit in making the 'unknown/uninitialised' values negative, and the 'known' values positive? That is, make `BATTERY_PRESCENCE_UNKNOWN = -2`